osf/models/files.py

```diff
@@ -445,7 +445,7 @@ def update(self, revision, data, user=None):
                 break
         else:
             # Insert into history if there is no matching etag
-            if data.get('modified'):
+            if data.get('modified') and None not in [item['modified'] for item in self.history]:
```

Comment:
Okay, so if any item anywhere in the history has a modified date of `None` then the new item goes to the bottom of the list? This is definitely better than erroring, but I'm not sure if it's right. Might be worth having a quick discussion with @felliott to see if there are any other potential options, since it's mostly about what can and can't actually be done.